construct user profile page with event list and deletion

98sean98 opened the pull request:
Closes #15 

`deleteEvent` mutation needs to be tested. for some reason I couldn't get my backend to return correct responses, possibly because I'm running the development-setup version

hopefully could get working event mutations in phoenix asap

98sean98 committed:
construct useDataFetching custom hook to fetch me and events data

98sean98 committed:
#15: build profile picture using media card

98sean98 committed:
#15: create a simple event list with delete button

98sean98 committed:
remove unnecessary redirect to login in useDataFetching hook

98sean98 committed:
fix useDataFetching edge cases where id token becomes invalid

welvin21 committed:
#15: add route to user profile; initialise UserProfile component

98sean98 committed:
#15: refactor user profile components

98sean98 committed:
#15: link to /user route in header

98sean98 committed:
centralise events data fetching with useDataFetching hook; set fetch me data in App component to network-only

remove redirect to login code in useDataFetching as top level App component already checks for me data validity

98sean98 committed:
fix EventForm bugs

move Modal component from /homepage to /shared directory

98sean98 committed:
#15: implement deleteEvent mutation

add shared Modal component

welvin21 commented on src/components/routes/homepage/Homepage.js:
```diff
@@ -1,22 +1,16 @@
-  if (eventsError) console.log(eventsError);
-
-  const { events } = eventsData;
+  const { me, events } = data;
```
Could you help removing `me` here, it's not actually being used inside `Events`. I did that because initially I thought deletion will be done on `Homepage`

welvin21 commented on src/components/routes/user-profile/Profile.js:
```diff
@@ -0,0 +1,43 @@
+      <CardActionArea>
+        <CardMedia
+          className={classes.media}
+          image={image_url ? image_url : placeholder}
```
let's use another placeholder for profile picture maybe, [this](https://www.google.com/search?q=profile+no+picture&tbm=isch&safe=strict&chips=q:profile+no+picture,g_1:facebook:FNjqG_fsEVY%3D&rlz=1C5CHFA_enHK859HK859&safe=strict&hl=en&ved=2ahUKEwjZnKDXlq3oAhWdxYsBHfjnAvkQ4lYoA3oECAEQGg&biw=1440&bih=821) might work 😂

welvin21 commented on src/components/routes/user-profile/filterEvents.js:
```diff
@@ -0,0 +1,3 @@
+export const filterEvents = (events, username) => {
+  return events.filter(event => event.organiser.username === username);
```
this could go to the `functions` directory. And we could destructure as well, `({ organiser: {username: organiserUsername} }) => organiserUsername === username`

98sean98 committed:
#15: make requested changes

remove unneeded me data in Homepage
destructure filterEvents function and move this to common functions directory
add empty profile pic placeholder

welvin21 commented:
@98sean98 I'll help rebasing to currently changed master

welvin21 commented:
@98sean98 i merged `bugs` to `master` a few minutes ago, dont forget to rebase 😂
